### PR TITLE
[ENG-5101] Add the ability to save and quasi-view a cedar record

### PR DESCRIPTION
**Pull request by bp-cos**
-   Ticket: [ENG-5101]
-   Feature flag: cedar

## Purpose

Add the ability to create and save a cedar record.

## Summary of Changes

Updates to existing files and extensive work in mirage.

## Screenshot(s)

N/A

## Side Effects

None

## QA Notes

Test away

[ENG-5101]: https://openscience.atlassian.net/browse/ENG-5101?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by adlius — CHANGES_REQUESTED** [review]
Some comments. Overall looks good.

File: lib/osf-components/addon/components/cedar-metadata-editor/component.ts [hunk 1]
```diff
@@ -1,23 +1,39 @@
...
+import Store from '@ember-data/store';
+import { inject as service } from '@ember/service';
 import Component from '@glimmer/component';
 import config from 'ember-osf-web/config/environment';
+import CedarMetadataRecordModel from 'ember-osf-web/models/cedar-metadata-record';
 
 const { cedarConfig } = config;
 
 interface Args {
     instanceObject: any;
```
[the remark this answers is not in the document]
Comment (adlius) [reply lib/osf-components/addon/components/cedar-metadata-editor/component.ts]: Indeed it was. When I wrote it I expect the whole component to just take in the JSON-LD blob as the `instanceObject` parameter. Now that it changes it makes sense to also change the name of the parameters.

File: lib/osf-components/addon/components/cedar-metadata-editor/component.ts [hunk 2]
```diff
@@ -1,23 +1,46 @@
...
+            record = this.store.createRecord('cedar-metadata-record');
+            record.template = this.args.cedarMetadataTemplate;
+            record.target = this.args.target;
+        }
+
+        // eslint-disable-next-line
+        // @ts-ignore
+        record.metadata = cee.currentMetadata;
+        await record.save().then(() => {
+            alert('this is for Futa - $5 - now saved');
```
[the remark this answers is not in the document]
Comment (futa-ikeda) [reply lib/osf-components/addon/components/cedar-metadata-editor/component.ts:41]: Interesting. Why is try/catch a smell? Are there better mechanisms to let users know when saving has failed?

File: app/guid-node/metadata/-components/metadata-tabs/template.hbs [hunk 3]
```diff
@@ -16,8 +16,8 @@
             {{did-insert this.didRenderList}}
         >
             <tablist.tab>{{t 'metadata.main-tab'}}</tablist.tab>
-            {{#each this.templates as |template| }}
-                <tablist.tab>{{template.metadata.name}}</tablist.tab>
+            {{#each this.cedarMetadataRecords as |cedarMetadataRecord| }}
```
[the remark this answers is not in the document]
Comment (futa-ikeda) [reply app/guid-node/metadata/-components/metadata-tabs/template.hbs:19]: I was curious if we would ever run into a case where there would be no `tablist.tab` or `tab.tabPanel ` shown, but I realize that there will always be at least one of those shown, so we don't need to worry about `this. cedarMetadataRecords` being an empty list.

As an aside, it looks like [JSON API says for empty has-many relationships](https://jsonapi.org/format/#fetching-relationships), it should return a collection of length 0

**Review by futa-ikeda — APPROVED** [review]
💸

**Pull request merged**